File: DataStructure/InterviewQuestion/BlockPair.cs
using System.Collections;

namespace DataStructure.InterviewQuestion
{
    public class BlockPair
    {
        public bool IsPair(string input)
        {
            var stack = new Stack();

            foreach (var c in input.ToCharArray())
            {
                if (stack.Count > 0 && (char)stack.Peek() == c)
                {
                    stack.Pop();
                }
                else
                {
                    stack.Push(c);
                }
            }

            return stack.Count == 0;
        }
        
    }
}

File: DataStructure/InterviewQuestion/JsPractice/BlockPair.js
function IsBlockPair(input) {

    var s = [];

    input.split("").forEach(function(c) {
        if (s.length > 0 && s[s.length - 1] == c) {
            s.pop();
        } else {
            s.push(c);
        }
    });

    return s.length == 0;
}